src/Uno.UI.Runtime.Skia.Wpf/Extensions/WpfNativeWebView.cs

```diff
@@ -0,0 +1,117 @@
+﻿#nullable enable
+
+extern alias WpfWebView;
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.WebView2.Core;
+using Uno.UI.Xaml.Controls;
+
+namespace Uno.UI.Runtime.Skia.Wpf.Extensions;
+
+internal sealed class WpfNativeWebView : INativeWebView
+{
+	private WpfWebView.Microsoft.Web.WebView2.Wpf.WebView2 _nativeWebView;
+	private CoreWebView2 _coreWebView2;
+	private List<Func<Task>> _actions = new();
+
+	public WpfNativeWebView(WpfWebView.Microsoft.Web.WebView2.Wpf.WebView2 nativeWebView, CoreWebView2 coreWebView2)
+	{
+		_coreWebView2 = coreWebView2;
+		_nativeWebView = nativeWebView;
+		nativeWebView.CoreWebView2InitializationCompleted += this.NativeWebView_CoreWebView2InitializationCompleted;
+		nativeWebView.EnsureCoreWebView2Async();
```

Comment:
Our API also exposes `EnsureCoreWebView2Async`, and users are required to call it anyway when they use `WebView2`, so I think it is safe to move this call there instead and that should simplify the logic here. It means that `WebView(1)` will not work on WPF as `EnsureCoreWebView2Async` is not there, but I think that is not a big deal

Reply:
It will be a bit problematic to do so.

We only create the INativeWebView in apply template. If we want to tie things up with our existing EnsureCoreWebView2Async, it means the task can only be completed after apply template, which doesn't feel right.

The current implementation is also not ideal 😕 

Let me know if you have some ideas.

Reply:
I see the problem. We will probably need to refactor it a bit to be to create the instance before onApplyTemplate. But that can be done outside this PR